app/controllers/general.php

```diff
@@ -446,6 +448,20 @@ function router(App $utopia, Database $dbForPlatform, callable $getProjectDB, Sw
                 requestTimeout: 30
             );
 
+
+            // Branded banner for previews
+            $transformation = new Transformation(new Preview($executionResponse['body']));
+            if ($type === 'deployment' && $transformation->isValid($executionResponse['headers'])) {
```

Comment:
I would expect the `isValid` method to be protected and used inside the transform method. After all, if the transformation is not valid you shouldn't allow it in any way. This way you can't enforce dev will write the code this way every time they need to transform something.